src/passes/transform/cse.cpp

```diff
@@ -0,0 +1,174 @@
+#include "coreir/passes/transform/cse.h"
+
+using namespace std;
+using namespace CoreIR;
+
+// Desired CoreIR iterators/functional methods
+// breadth-first tree iterator on wireables
+// depth-first tree iterator on wireables
+// leaf-iterator on wireables (Only single bits)
+// breadth-first iteration on types
+// depth-first iterator on types
+// leaf iterator on types 
+// 'All' operator with a predicate
+// 'Any' operator with a predicate
+// 'Map' operator
+// 'Filter' operator (both a modify and a return new)
+
+
+//Some possible (untested) implementations of some of the functions
```

Comment:
Space between `//` and first character of comment. End full line comments in `.`. Make sure to wrap at 80 chars.  Same for all comments below.